### PR TITLE
[Fixes #15953] Implement assert.expect in each unit test

### LINKED ISSUE #15953: Unit tests: Implement assert.expect in each unit test
And require every future unit test to also do so.

See https://github.com/twbs/bootstrap/pull/15593/files#r25565051 as to why.

/cc @cvrebert

### PR DESCRIPTION
**Pull request by kkirsche**
[Fixes #15953] Implement `assert.expect` in each unit test

### CODE REVIEW

**Comment by twbs-savage** [comment]
**Tests passed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **CONFIRMED**

Commit: 26a8bb83efc42343e2345c6715c52a84a3252317
Build details: https://travis-ci.org/twbs-savage/bootstrap/builds/52628053

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by hnrch02** [comment]
I imagined `assert.expect` being the very first line of every test.

**Comment by kkirsche** [comment]
@hnrch02 I moved it to second, after `done()` to follow some of the asserts that were already there such as [this](https://github.com/twbs/bootstrap/blob/049e86f47e90b35eeaf97d6b735f2fd3d62c0d1c/js/tests/unit/collapse.js#L142)

**Comment by hnrch02** [comment]
I see. IMO, it's more consistent with the non-async tests if it is the first line.

**Comment by kkirsche** [comment]
@hnrch02 I'll move them all to the top for consistency, all the non-async are already at the top though unless they have a browser compatibility check with an `assert.expect(0)` in it.

**Comment by hnrch02** [comment]
Thanks :+1:

**Comment by kkirsche** [comment]
Fixed @hnrch02 :smiley:

**Comment by twbs-savage** [comment]
**Tests passed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **CONFIRMED**

Commit: cf2d5543f3d7f2b412fad382df198f6755fc4bbf
Build details: https://travis-ci.org/twbs-savage/bootstrap/builds/52629724

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by cvrebert** [comment]
We should enable the `requireExpects` QUnit config option (mentioned on http://api.qunitjs.com/QUnit.config/).

**Comment by kkirsche** [comment]
@cvrebert Enabled :smiley:

**Comment by twbs-savage** [comment]
**Tests passed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **CONFIRMED**

Commit: 3f674a60c2b90723a975339e85dc8881d76901b6
Build details: https://travis-ci.org/twbs-savage/bootstrap/builds/52740523

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by twbs-savage** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: 0da5b08ce92e5d012c6fcc342d3e37ee63872c99
Build details: https://travis-ci.org/twbs-savage/bootstrap/builds/52745834

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by kkirsche** [comment]
Not sure what's going on with Sauce Labs. This shouldn't bust anything with them...

**Comment by twbs-savage** [comment]
**Tests passed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **CONFIRMED**

Commit: 2f8a249122dba4ae1c433664715af4853bec1cfb
Build details: https://travis-ci.org/twbs-savage/bootstrap/builds/52746442

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

File: js/tests/unit/phantom.js [hunk 1]
```diff
@@ -13,6 +13,8 @@
   QUnit.config.reorder = false
   // Run tests serially, not in parallel.
   QUnit.config.autorun = false
+  // Require assert.expect in each test.
```
Comment (cvrebert) [inline js/tests/unit/phantom.js]: We want this enabled everywhere, not just for PhantomJS, so this should go in index.html instead.

**Comment by cvrebert** [comment]
Other than that, LGTM.

File: js/tests/unit/phantom.js (hunk 1 above)
Comment (kkirsche) [inline js/tests/unit/phantom.js]: Moved from `phantom.js` to `index.html`. Sorry about that!

**Comment by twbs-savage** [comment]
**Tests passed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **CONFIRMED**

Commit: 4febcb4b492c322c71bdef579cc4ad99ea86e150
Build details: https://travis-ci.org/twbs-savage/bootstrap/builds/52895764

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by hnrch02** [comment]
Thanks!

**Comment by kkirsche** [comment]
My pleasure. Thanks for the reviews and feedback :+1: Helps me learn more about the project and the architecture.

**Pull request merged**